pages/box-office/_performanceId/sell/complete.vue

```diff
@@ -34,41 +58,96 @@ export default {
       type: Booking,
     },
   },
-  async mounted() {
+  data() {
+    return {
+      checkedIn: false,
+      currentTime: new Date(),
+    }
+  },
+  computed: {
+    performanceDoorsDiffMinutes() {
+      return (
+        (Date.parse(this.booking.performance.doorsOpen) - this.currentTime) /
+        60000
+      )
+    },
+    autoCheckIn() {
+      // Performance doors are opening within 15 minutes
+      return this.performanceDoorsDiffMinutes <= 15
+    },
+  },
+  mounted() {
     if (!this.booking.reference) return this.$router.push('../')
 
-    try {
-      await performMutation(
-        this.$apollo,
-        {
-          mutation: CheckInTickets,
-          variables: {
-            reference: this.booking.reference,
-            performanceId: this.booking.performance.id,
-            tickets: this.booking.tickets.map((ticket) => {
-              return {
-                ticketId: ticket.id,
-              }
-            }),
-          },
-        },
-        'checkInBooking'
-      )
-      successToast.fire({
-        timer: 4000,
-        title: 'Tickets automatically checked in',
-      })
-    } catch (e) {
-      errorToast.fire({
-        title: 'Unable to check in tickets automatically',
-      })
-    }
+    if (this.autoCheckIn) this.checkInTickets()
   },
   beforeDestroy() {
     // Remove stored booking ID
     this.$store.commit('box-office/SET_IN_PROGRESS_BOOKING_ID', null)
   },
   methods: {
+    async checkInTickets() {
+      try {
+        await performMutation(
+          this.$apollo,
+          {
+            mutation: CheckInTickets,
+            variables: {
+              reference: this.booking.reference,
+              performanceId: this.booking.performance.id,
+              tickets: this.booking.tickets.map((ticket) => {
+                return {
+                  ticketId: ticket.id,
+                }
+              }),
+            },
+          },
+          'checkInBooking'
+        )
+        this.checkedIn = true
+        let message = 'Tickets checked in'
+        if (this.autoCheckIn) {
+          message = 'Tickets automatically checked in'
+        }
```

Comment:
```suggestion
        let message = this.autoCheckIn ? 'Tickets automatically checked in' : 'Tickets checked in'
```

Could move inline to line 114 if you wanted

Reply:
tried it and swal didnt print the evaluation of the function